src/unix.rs

```diff
@@ -130,6 +132,18 @@ impl MmapInner {
         }
     }
 
+    pub fn set_protection(&mut self, prot: Protection) -> io::Result<()> {
+        let result = unsafe { libc::mprotect(self.ptr,
```

Comment:
You can't go from read+write to read+execute since the latter isn't less restrictive than the former, we'd need `Protection::ReadWriteExecute` to do that, and I'm not sure if it's that useful. But I'll add a test to make sure `set_protection` works when there's an offset.